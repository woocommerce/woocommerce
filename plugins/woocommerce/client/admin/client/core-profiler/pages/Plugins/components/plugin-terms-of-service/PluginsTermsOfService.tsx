/**
 * External dependencies
 */
import { sprintf, _n } from '@wordpress/i18n';
import interpolateComponents from '@automattic/interpolate-components';
import { Link } from '@woocommerce/components';
import { ExtensionList } from '@woocommerce/data';
/**
 * Internal dependencies
 */
import { joinWithAnd, composeListFormatParts } from '../../Plugins';

export const PluginsTermsOfService = ( {
	selectedPlugins,
}: {
	selectedPlugins: ExtensionList[ 'plugins' ];
} ) => {
	const pluginsWithTOS = selectedPlugins.filter( ( plugin ) =>
		[
			'jetpack',
			'woocommerce-services:shipping',
			'woocommerce-services:tax',
		].includes( plugin.key )
	);

	if ( ! pluginsWithTOS.length ) {
		return null;
	}

	return (
		<p className="woocommerce-profiler-plugins-jetpack-agreement">
			{ interpolateComponents( {
				mixedString: sprintf(
					/* translators: %s: a list of plugins, e.g. Jetpack */
					_n(
						'By installing %s plugin for free you agree to our {{link}}Terms of Service{{/link}}.',
						'By installing %s plugins for free you agree to our {{link}}Terms of Service{{/link}}.',
						pluginsWithTOS.length,
						'woocommerce'
					),
					joinWithAnd(
						pluginsWithTOS.map( ( plugin ) => plugin.name )
					)
						.map( composeListFormatParts )
						.join( '' )
				),
				components: {
					span: <span />,
					link: (
						<Link
							href="https://wordpress.com/tos/"
							target="_blank"
							type="external"
						/>
					),
				},
			} ) }
		</p>
	);
};
