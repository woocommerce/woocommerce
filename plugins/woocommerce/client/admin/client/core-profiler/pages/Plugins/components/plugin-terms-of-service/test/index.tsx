// write tests for PluginTermsOfService

/**
 * External dependencies
 */
import { render, screen } from '@testing-library/react';
import React from 'react';

/**
 * Internal dependencies
 */
import { PluginsTermsOfService } from '../PluginsTermsOfService';

describe( 'PluginsTermsOfService', () => {
	it( 'should not render anything when no plugins with TOS are selected', () => {
		const { container } = render(
			<PluginsTermsOfService
				selectedPlugins={ [
					{
						key: 'woocommerce-payments',
						name: 'WooCommerce Payments',
						description: '',
						image_url: '',
						manage_url: '',
						is_built_by_wc: false,
						is_visible: false,
					},
				] }
			/>
		);
		expect( container ).toBeEmptyDOMElement();
	} );

	it( 'should render TOS message when Jetpack is selected', () => {
		render(
			<PluginsTermsOfService
				selectedPlugins={ [
					{
						key: 'jetpack',
						name: 'Jetpack',
						description: '',
						image_url: '',
						manage_url: '',
						is_built_by_wc: false,
						is_visible: false,
					},
				] }
			/>
		);
		expect(
			screen.getByText( /plugin for free you agree to our/i )
		).toMatchSnapshot();
		expect( screen.getByText( 'Jetpack' ) ).toBeInTheDocument();
		expect( screen.getByText( 'Terms of Service' ) ).toBeInTheDocument();
	} );

	it( 'should render TOS message with multiple plugins when more than one plugin with TOS is selected', () => {
		render(
			<PluginsTermsOfService
				selectedPlugins={ [
					{
						key: 'jetpack',
						name: 'Jetpack',
						description: '',
						image_url: '',
						manage_url: '',
						is_built_by_wc: false,
						is_visible: false,
					},
					{
						key: 'woocommerce-services:shipping',
						name: 'WooCommerce Shipping',
						description: '',
						image_url: '',
						manage_url: '',
						is_built_by_wc: false,
						is_visible: false,
					},
					{
						key: 'woocommerce-services:tax',
						name: 'WooCommerce Tax',
						description: '',
						image_url: '',
						manage_url: '',
						is_built_by_wc: false,
						is_visible: false,
					},
				] }
			/>
		);
		expect( screen.getByText( /By installing/ ) ).toMatchSnapshot();
	} );
} );
