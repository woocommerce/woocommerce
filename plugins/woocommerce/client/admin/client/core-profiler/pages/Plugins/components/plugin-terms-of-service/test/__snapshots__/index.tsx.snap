// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginsTermsOfService should render TOS message when Jetpack is selected 1`] = `
<p
  class="woocommerce-profiler-plugins-jetpack-agreement"
>
  By installing 
  <span>
    Jetpack
  </span>
   plugin for free you agree to our 
  <a
    data-link-type="external"
    href="https://wordpress.com/tos/"
    target="_blank"
  >
    Terms of Service
  </a>
  .
</p>
`;

exports[`PluginsTermsOfService should render TOS message with multiple plugins when more than one plugin with TOS is selected 1`] = `
<p
  class="woocommerce-profiler-plugins-jetpack-agreement"
>
  By installing 
  <span>
    Jetpack
  </span>
  , 
  <span>
    WooCommerce Shipping
  </span>
  , and 
  <span>
    WooCommerce Tax
  </span>
   plugins for free you agree to our 
  <a
    data-link-type="external"
    href="https://wordpress.com/tos/"
    target="_blank"
  >
    Terms of Service
  </a>
  .
</p>
`;
